docs/resources/instance.md

```diff
@@ -141,12 +143,19 @@ The `file` block supports:
 * `create_directories` - *Optional* - Whether to create the directories leading
 	to the target if they do not exist.
 
-The `exec` block supports:
+The `execs` map elements support the following attributes:
 
 * `command` - **Required** - The command to be executed and its arguments, if any (list of strings).
 
-* `triggers` - *Optional* - A list of arbitrary strings that, when changed, will force the command
-  to be rerun.
+* `enabled` - *Optional* - Boolean indicating whether the command should be executed.
+  Defaults to `true`.
+
+* `trigger` - *Optional* - Determines when the command should be executed.
+  Available values are:
+  + `on_change` (default) - Executes the command whenever the instance resource changes.
+  + `on_start` - Executes the command on instance start. Note that the command will **not** be executed
+    if the instance is started outside of Terraform.
+  + `once` - Executes the command only once.
 
 * `environment` - *Optional* - Map of additional environment variables.
```

Comment:
Are the environment variables injected into the container or are only available for the command?

Reply:
Environment variables specified within the exec command are exclusive to that command and are not accessible elsewhere. To apply environment variables at the instance level, they must be defined in the `config`. 

For example:
```hcl
resource "lxd_instance" "inst" {
  name      = "c1"
  image     = "ubuntu-daily:22.04"

  config = {
    "environment.MY_ENV" = "value123"
  }

  execs = {
    "1" = {
      command       = ["sh", "-c", "echo $MY_ENV"]
      record_output = true
    }
  }
}

output "exec-output" {
  value = lxd_instance.inst.execs["1"].stdout # "value123\n"
}
```

LXD docs: https://documentation.ubuntu.com/lxd/en/latest/instance-exec/#environment

Reply:
Great Thank you :)